Percy/providers/GenericProvider.cs

```diff
@@ -76,20 +79,167 @@ internal void SetDebugUrl(string debugUrl)
       this.debugUrl = debugUrl;
     }
 
-    public virtual String Screenshot(String name, String deviceName, int statusBarHeight, int navBarHeight,
-        String orientation, Boolean fullScreen, bool fullPage, int? screenLengths, String platformVersion = null)
+    public virtual String Screenshot(String name, ScreenshotOptions options ,String platformVersion = null)
     {
       this.metadata = MetadataHelper.Resolve(
         percyAppiumDriver,
-        deviceName,
-        statusBarHeight,
-        navBarHeight,
-        orientation,
+        options.DeviceName,
+        options.StatusBarHeight,
+        options.NavBarHeight,
+        options.Orientation,
         platformVersion
       );
       var tag = GetTag();
-      var tiles = CaptureTiles(fullScreen, fullPage, screenLengths);
-      return CliWrapper.PostScreenshot(name, tag, tiles, debugUrl);
+      var ignoredElementLocation = IgnoredElementsLocation(options);
+      var tiles = CaptureTiles(options);
+      return CliWrapper.PostScreenshot(name, tag, tiles, debugUrl, ignoredElementLocation);
+    }
+
+    public JObject IgnoredElementsLocation(ScreenshotOptions options)
+    {
+      var ignoredElementsArray = new JArray();
+      IgnoreLocationByXpaths(ignoredElementsArray, options.Xpaths);
+      IgnoreLocationByIds(ignoredElementsArray, options.AccessibilityIds);
+      IgnoreLocationByElement(ignoredElementsArray, options.AppiumElements);
+      AddCustomIgnoreLocation(ignoredElementsArray, options.CustomIgnoreRegions);
+
+      var ignoredElementsLocations = JObject.FromObject(new
+      {
+        ignoreElementsData = ignoredElementsArray
+      });
+      
+      return ignoredElementsLocations;
+    }
+
+    public JObject IgnoreElementObject(String selector, Point location, Size size)
+    {
+      return JObject.FromObject(new
+      {
+        selector = selector,
+        co_ordinates = new
+        {
+          top = location.Y,
+          bottom = location.Y + size.Height,
+          left = location.X,
+          right = location.X + size.Width
+        }
+      });     
+    }
+
+    public void IgnoreLocationByXpaths(JArray ignoredElementsArray, List<String> xpaths)
+    {
+      var index = 0;
+      foreach (var xpath in xpaths)
+      {
+        try
+        {
+          var element =  percyAppiumDriver.FindElementByXPath(xpath);
+          
+          var location = element.Location;
+          var size = element.Size;
+          var selector = string.Format("xpath {0} {1}", index, xpath);
```

Comment:
```suggestion
          var selector = string.Format("xpath: {1}", xpath);
```
Dont need index where values have meaning